html/js/product-preferences.js

```diff
@@ -23,6 +23,32 @@ function get_user_product_preferences() {
 	return user_product_preferences;
 }
 
+function getFlavorFromHostname() { // get the flavor by the url 
```

Comment:
This probably works, but I would prefer that we pass the the flavor from the server, so that we don't have to update the JS code when we add new flavors, or change preferences for OPF or OPFF.

See the existing mechanism to pass variables from server to JS here: https://github.com/openfoodfacts/openfoodfacts-server/blob/main/lib/ProductOpener/Display.pm#L4435